src/components/PostsList.tsx

```diff
@@ -1,36 +1,77 @@
-import React from 'react';
-
-export const PostsList: React.FC = () => (
-  <div data-cy="PostsList">
-    <p className="title">Posts:</p>
-
-    <table className="table is-fullwidth is-striped is-hoverable is-narrow">
-      <thead>
-        <tr className="has-background-link-light">
-          <th>#</th>
-          <th>Title</th>
-          {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-          <th> </th>
-        </tr>
-      </thead>
+import React, { useState } from 'react';
+import { Post } from '../types/Post';
+import cl from 'classnames';
 
-      <tbody>
-        <tr data-cy="Post">
+interface Props {
+  posts: Post[];
+  onActivePost: (v: Post | null) => void;
+  onShowSideBar: (v: boolean) => void;
+}
+
+export const PostsList: React.FC<Props> = ({
+  posts,
+  onActivePost,
+  onShowSideBar,
+}) => {
+  const [activePost, setActivePost] = useState<Post | null>(null);
+  const [showClose, setShowClose] = useState(false);
+
+  const handleChangeButton = (post: Post) => {
+    if (post.id === activePost?.id) {
+      setShowClose(!showClose);
+      onShowSideBar(showClose ? false : true);
```

Comment:
The logic for toggling `showClose` and `onShowSideBar` could be simplified for better readability. Consider using a single state to manage the sidebar visibility.